src/libpriv/rpmostree-origin.cxx

```diff
@@ -538,19 +533,8 @@ rpmostree_origin_set_regenerate_initramfs (RpmOstreeOrigin *origin, gboolean reg
 void
 rpmostree_origin_set_override_commit (RpmOstreeOrigin *origin, const char *checksum)
 {
-  if (checksum != NULL)
-    {
-      g_key_file_set_string (origin->kf, "origin", "override-commit", checksum);
-    }
-  else
-    {
-      g_key_file_remove_key (origin->kf, "origin", "override-commit", NULL);
-    }
-
-  g_free (origin->cached_override_commit);
-  origin->cached_override_commit = g_strdup (checksum);
-
   (*origin->treefile)->set_override_commit (checksum ?: "");
+  sync_origin (origin);
```

Comment:
Do we still need this call though?

Reply:
I nuked `sync_origin`! I initially did want it because it's potentially dangerous to have a stale `origin` cache. But in fact, after construction there is now no way to extract the `GKeyFile` from an `RpmOstreeOrigin` so this is not a concern.

Reply:
Oh right OK, I remember now why we do really need this. If the `origin` cache goes out of sync, calling any setter not yet ported over which calls `sync_treefile` will make that outdated cache canonical.

Essentially, temporarily while we transition over, we need to maintain across all setters the invariant that the treefile and keyfile are always the same. We use `sync_treefile` on setters not yet migrated, and `sync_origin` on setters migrated.